backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -899,8 +910,9 @@ public ExperimentRun deleteExperimentRunTags(
   public void logObservations(String experimentRunId, List<Observation> observations)
       throws InvalidProtocolBufferException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      Transaction transaction = session.beginTransaction();
```

Comment:
Does this Transaction automatically rollback on an exception?  I see the explicit commit, but no explicit rollback.

Reply:
when we are called `transaction.commit()` then it will reflect on DB and before that exception throw then no need to do a rollback. In our entire project, we have not used rollback anywhere.

Reply:
👍